### PR TITLE
JBPM-8084 - Allow to use Kie Server rendered forms in workbench

### PR DESCRIPTION
**Pull request by mswiderski**
depends on https://github.com/kiegroup/droolsjbpm-integration/pull/1671

to give it a go just start server with -Dorg.jbpm.wb.forms.renderer.ext=true

### CODE REVIEW

**Comment by mswiderski** [comment]
Jenkins execute full downstream build

File: jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormRendererProxyServlet.java [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        String pathInfo = req.getPathInfo();
+        
+        if (pathInfo == null) {
+            throw new IllegalArgumentException("Missing path information");
+        }
+        
+        String body = req.getReader().lines().collect(Collectors.joining());
+        
+        String containerId = extractValue(containerPattern, pathInfo);
+        if (pathInfo.contains("cases")) {
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormRendererProxyServlet.java]: nice :)

File: jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormServiceEntryPointImpl.java [hunk 2]
```diff
@@ -68,14 +72,101 @@ public FormServiceEntryPointImpl(Instance<FormProvider<? extends FormRenderingSe
...
+        } else {
+            return getEmbeddedFormDisplayProcess(serverTemplateId, domainId, processId);
+        }
+    }
+    
+    /*
+     * KieServer based form rendering
+     */
+    
+    protected FormRenderingSettings getKieServerFormDisplayProcess(String serverTemplateId,
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormServiceEntryPointImpl.java]: small formatting issue here... overall if you can also remove some of the extra empty lines.

File: jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieServerFormsHumanTaskDisplayer.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                                                    true);
+        inlineFrame.getElement().getStyle().setBorderWidth(0,
+                                                           Style.Unit.PX);
+        inlineFrame.getElement().getStyle().setPosition(Position.ABSOLUTE);
+        inlineFrame.getElement().getStyle().setTop(0, Unit.PX);
+        inlineFrame.getElement().getStyle().setLeft(0, Unit.PX);
+        
+        inlineFrame.setUrl(renderingSettings.getUrl());
+        
+        FlowPanel div = new FlowPanel();
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieServerFormsHumanTaskDisplayer.java]:
```suggestion
        FlowPanel div = GWT.create(FlowPanel.class);
```

File: jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieServerFormsStartProcessDisplayer.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                                                    true);
+        inlineFrame.getElement().getStyle().setBorderWidth(0,
+                                                           Style.Unit.PX);
+        inlineFrame.getElement().getStyle().setPosition(Position.ABSOLUTE);
+        inlineFrame.getElement().getStyle().setTop(0, Unit.PX);
+        inlineFrame.getElement().getStyle().setLeft(0, Unit.PX);
+        
+        inlineFrame.setUrl(renderingSettings.getUrl());
+        
+        FlowPanel div = new FlowPanel();
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieServerFormsStartProcessDisplayer.java]:
```suggestion
        FlowPanel div = GWT.create(FlowPanel.class);
```

File: jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieServerFormsStartProcessDisplayer.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+    
+    public void notifyAboutStartProcess(String id) {
+        
+        Long processInstanceId = Long.parseLong(id);
+        newProcessInstanceEvent.fire(new NewProcessInstanceEvent(serverTemplateId,
+                                                                 deploymentId,
+                                                                 processInstanceId,
+                                                                 processDefId,
+                                                                 processName,
+                                                                 1));
```
Comment (cristianonicolai) [inline jbpm-wb-forms/jbpm-wb-forms-client/src/main/java/org/jbpm/workbench/forms/client/display/KieServerFormsStartProcessDisplayer.java]: can you pass null in here for the status, I recently refactored similar code, and status should not be part of the event.

**Comment by mswiderski** [comment]
@cristianonicolai all suggestions applied, thanks

**Comment by jsoltes** [comment]
jenkins please execute full downstream build

File: jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormServiceEntryPointImpl.java [hunk 6]
```diff
@@ -68,14 +72,98 @@ public FormServiceEntryPointImpl(Instance<FormProvider<? extends FormRenderingSe
...
+                                                                   String domainId,
+                                                                   String processId,
+                                                                   boolean isDynamic) {
+
+        try {
+            Map<String, Object> valuesMap = new HashMap<String, Object>();
+            valuesMap.put(RestURI.CONTAINER_ID, domainId);
+            valuesMap.put(RestURI.PROCESS_ID, processId);
+            valuesMap.put(RestURI.CASE_DEF_ID, processId);
```
Comment (MarianMacik) [inline jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormServiceEntryPointImpl.java]: I don't see this used anywhere.

File: jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormServiceEntryPointImpl.java [hunk 7]
```diff
@@ -68,14 +72,98 @@ public FormServiceEntryPointImpl(Instance<FormProvider<? extends FormRenderingSe
...
+    }
+    
+    protected FormRenderingSettings getKieServerFormDisplayTask(String serverTemplateId,
+                                                                String domainId,
+                                                                long taskId) {
+
+        try {
+            Map<String, Object> valuesMap = new HashMap<String, Object>();
+            valuesMap.put(RestURI.CONTAINER_ID, domainId);
+            valuesMap.put(RestURI.TASK_INSTANCE_ID, taskId);
```
Comment (MarianMacik) [inline jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormServiceEntryPointImpl.java]: Same here.

File: jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormServiceEntryPointImpl.java [hunk 8]
```diff
@@ -56,10 +59,11 @@
 @ApplicationScoped
 public class FormServiceEntryPointImpl extends AbstractKieServerService implements FormServiceEntryPoint {
 
-    private static final Logger logger = LoggerFactory.getLogger(FormServiceEntryPointImpl.class);
-
+    private static final Logger logger = LoggerFactory.getLogger(FormServiceEntryPointImpl.class);  
+    
+    private boolean kieServerFormRenderer = KieServerUtils.isKieServerRenererEnabled();
```
Comment (MarianMacik) [inline jbpm-wb-forms/jbpm-wb-forms-backend/src/main/java/org/jbpm/workbench/forms/display/backend/FormServiceEntryPointImpl.java]: typo in `isKieServerRenererEnabled()`

**Comment by mswiderski** [comment]
@MarianMacik changes applied

**Pull request merged**